donut/modules/voting/routes.py

```diff
@@ -0,0 +1,264 @@
+from datetime import date, datetime, timedelta
+import json
+import flask
+
+from . import blueprint, helpers
+from donut.modules.core.helpers import get_member_data
+
+
+@blueprint.route('/1/surveys')
+def list_surveys():
+    user_id = helpers.get_user_id(flask.session.get('username'))
+    active_surveys = helpers.get_public_surveys(user_id)
+    closed_surveys = helpers.get_closed_surveys(user_id)
+    return flask.render_template(
+        'list_surveys.html',
+        active_surveys=active_surveys,
+        closed_surveys=closed_surveys)
+
+
+@blueprint.route('/1/surveys/<access_key>/take')
+def take_survey(access_key):
+    survey = helpers.get_survey_data(access_key)
+    restrict_message = helpers.restrict_take_access(survey)
+    if restrict_message:
+        flask.flash(restrict_message)
+        return list_surveys()
+
+    questions_json = helpers.get_questions_json(survey['survey_id'], True)
+    user_id = helpers.get_user_id(flask.session['username'])
+    is_owner = survey['creator'] == user_id
+    return flask.render_template(
+        'take.html',
+        access_key=access_key,
+        **survey,
+        is_owner=is_owner,
+        NO=helpers.NO,
+        question_types=helpers.get_question_types(),
+        questions_json=questions_json)
+
+
+@blueprint.route('/1/surveys/make')
+def make_survey_form():
+    tomorrow = date.today() + timedelta(days=1)
```

Comment:
probably unnecessary to define defaults for these, since they're required for every survey and whoever is making the survey will enter them in anyways. so unless they're necessary to make the code work (i.e. variables can't be null) we should force whoever is making the survey to make sure they enter the dates in manually to make sure they are entering it correctly.